app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt

```diff
@@ -0,0 +1,810 @@
+    val readText = stringResource(R.string.read)
+    val startPage by rememberInVM {
+        EhDB.getReadProgressFlow(galleryInfo.gid)
+    }.collectAsState(0)
+    val readButtonText = if (startPage == 0) {
+        readText
+    } else {
+        stringResource(R.string.read_from, startPage + 1)
+    }
+    val downloadState by DownloadManager.collectDownloadState(galleryInfo.gid)
+    val downloadButtonText = when (downloadState) {
+        DownloadInfo.STATE_INVALID -> stringResource(R.string.download)
+        DownloadInfo.STATE_NONE -> stringResource(R.string.download_state_none)
+        DownloadInfo.STATE_WAIT -> stringResource(R.string.download_state_wait)
+        DownloadInfo.STATE_DOWNLOAD -> stringResource(R.string.download_state_downloading)
+        DownloadInfo.STATE_FINISH -> stringResource(R.string.download_state_downloaded)
+        DownloadInfo.STATE_FAILED -> stringResource(R.string.download_state_failed)
+        else -> error("Invalid DownloadState!!!")
+    }
+    fun onReadButtonClick() {
+        if (galleryDetail != null || downloadState != DownloadInfo.STATE_INVALID) {
+            navToReader(galleryInfo.findBaseInfo(), startPage)
+        }
+    }
+    fun onCategoryChipClick() {
+        val category = galleryInfo.category
+        if (category == EhUtils.NONE || category == EhUtils.PRIVATE || category == EhUtils.UNKNOWN) {
+            return
+        }
+        val lub = ListUrlBuilder(category = category)
+        navigate(lub.asDst())
+    }
+    fun onUploaderChipClick(galleryInfo: GalleryInfo) {
+        val uploader = galleryInfo.uploader
+        val disowned = uploader == "(Disowned)"
+        if (uploader.isNullOrEmpty() || disowned) {
+            return
+        }
+        val lub = ListUrlBuilder()
+        lub.mode = ListUrlBuilder.MODE_UPLOADER
+        lub.keyword = uploader
+        navigate(lub.asDst())
+    }
+
+    var showBottomSheet by remember { mutableStateOf(false) }
+
+    if (showBottomSheet && galleryDetail != null) {
+        ModalBottomSheet(
+            onDismissRequest = { showBottomSheet = false },
+            contentWindowInsets = { EmptyWindowInsets },
+        ) {
+            GalleryInfoBottomSheet(galleryDetail)
+        }
+    }
+
+    fun onGalleryInfoCardClick() {
+        showBottomSheet = true
+    }
+
+    val filterAdded = stringResource(R.string.filter_added)
+    fun showFilterUploaderDialog(galleryInfo: GalleryInfo) {
+        val uploader = galleryInfo.uploader
+        val disowned = uploader == "(Disowned)"
+        if (uploader.isNullOrEmpty() || disowned) {
+            return
+        }
+        launchIO {
+            awaitPermissionOrCancel {
+                Text(text = stringResource(R.string.filter_the_uploader, uploader))
+            }
+            Filter(FilterMode.UPLOADER, uploader).remember()
+            showSnackbar(filterAdded)
+        }
+    }
+    val onDownloadButtonClick = rememberLambda(galleryInfo) {
```

Comment:
We shuld investigate to remove this `rememberLambda` usage and definition, IIRC it is introduced with strongskipping mode, better to use @Dontmemorize